core/trino-main/src/main/java/io/trino/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -608,6 +610,30 @@ public PlanAndMappings visitUpdate(UpdateNode node, UnaliasContext context)
                     mapping);
         }
 
+        @Override
+        public PlanAndMappings visitMerge(MergeNode node, UnaliasContext context)
+        {
+            PlanAndMappings rewrittenSource = node.getSource().accept(this, context);
+            Map<Symbol, Symbol> mapping = new HashMap<>(rewrittenSource.getMappings());
+            SymbolMapper mapper = symbolMapper(mapping);
+
+            MergeNode rewrittenMerge = mapper.map(node, rewrittenSource.getRoot());
+
+            return new PlanAndMappings(rewrittenMerge, mapping);
+        }
+
+        @Override
+        public PlanAndMappings visitDeleteAndInsert(DeleteAndInsertNode node, UnaliasContext context)
+        {
+            PlanAndMappings rewrittenSource = node.getSource().accept(this, context);
+            Map<Symbol, Symbol> mapping = new HashMap<>(rewrittenSource.getMappings());
+            SymbolMapper mapper = symbolMapper(mapping);
+
+            DeleteAndInsertNode rewrittenMerge = mapper.map(node, rewrittenSource.getRoot());
```

Comment:
Rename variable `rewrittenMerge` to match the context.

Reply:
Renamed to `deleteAndInsertNode`.